src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/OpponentsSectionController.java

```diff
@@ -0,0 +1,276 @@
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.SessionAttribute;
+import org.springframework.web.bind.annotation.SessionAttributes;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+import uk.gov.laa.ccms.caab.bean.OpponentFormData;
+import uk.gov.laa.ccms.caab.bean.OrganisationSearchCriteria;
+import uk.gov.laa.ccms.caab.bean.validators.application.OpponentOrganisationValidator;
+import uk.gov.laa.ccms.caab.bean.validators.application.OrganisationSearchCriteriaValidator;
+import uk.gov.laa.ccms.caab.builders.DropdownBuilder;
+import uk.gov.laa.ccms.caab.exception.CaabApplicationException;
+import uk.gov.laa.ccms.caab.exception.TooManyResultsException;
+import uk.gov.laa.ccms.caab.model.OpponentRowDisplay;
+import uk.gov.laa.ccms.caab.model.OrganisationResultRowDisplay;
+import uk.gov.laa.ccms.caab.model.ResultsDisplay;
+import uk.gov.laa.ccms.caab.service.ApplicationService;
+import uk.gov.laa.ccms.caab.service.LookupService;
+import uk.gov.laa.ccms.caab.service.OpponentService;
+import uk.gov.laa.ccms.data.model.CommonLookupDetail;
+import uk.gov.laa.ccms.data.model.RelationshipToCaseLookupDetail;
+import uk.gov.laa.ccms.data.model.UserDetail;
+
+/**
+ * Controller for the application's opponents and other parties section.
+ */
+@Controller
+@RequiredArgsConstructor
+@Slf4j
+@SessionAttributes(value = {
+    ORGANISATION_SEARCH_CRITERIA,
+    ORGANISATION_SEARCH_RESULTS,
+    CURRENT_OPPONENT})
+public class OpponentsSectionController {
+
+  private final ApplicationService applicationService;
+
+  private final OpponentService opponentService;
+
+  private final LookupService lookupService;
+
+  private final OrganisationSearchCriteriaValidator searchCriteriaValidator;
+
+  private final OpponentOrganisationValidator opponentOrganisationValidator;
+
+
+  /**
+   * Provides an instance of {@link OrganisationSearchCriteria} for use in the model.
+   *
+   * @return A new instance of {@link OrganisationSearchCriteria}.
+   */
+  @ModelAttribute(ORGANISATION_SEARCH_CRITERIA)
+  public OrganisationSearchCriteria getOrganisationSearchCriteria() {
+    return new OrganisationSearchCriteria();
+  }
+
+  /**
+   * Handles the GET request to fetch and display the opponents and other parties for a specific
+   * application.
+   *
+   * @param applicationId The id of the application, retrieved from the session.
+   * @param model The Model object to add attributes to for the view.
+   * @param session The HttpSession object to add attributes to for the session.
+   * @return The name of the view to be rendered.
+   */
+  @GetMapping("/application/summary/opponents")
```

Comment:
Should this be application/opponents/summary ?

Reply:
I thought we were keeping a summary folder for the current snapshot of each application section. But can be moved if we're not doing that?

Reply:
looking at mine i chage proceeding and costs to:

@GetMapping("/application/proceedings-and-costs")

no summary at all 

But for proceedings directly i do:

@GetMapping("/application/proceedings/{proceeding-id}/summary")